src/schemas/Program/index.js

```diff
@@ -388,66 +363,78 @@ const resolvers = {
 				genomicDonors: 0,
 			};
 
-			try {
-				const createResponse = await programService.createProgram(program, egoToken);
-				return resolvePrivateSingleProgram(egoToken, program.shortName);
-			} catch (err) {
-				const GRPC_INVALID_ARGUMENT_ERROR_CODE = 3;
-				if (err.code === GRPC_INVALID_ARGUMENT_ERROR_CODE) {
-					//this just wraps it into standard apollo semantics
-					throw new UserInputError(err);
-				} else {
-					throw new ApolloError(err);
-				}
-			}
+			const createResponse = await programService.createProgram(program, egoToken);
+			return resolvePrivateSingleProgram(egoToken, program.shortName);
 		},
 
 		updateProgram: async (obj, args, context, info) => {
+			// extract information from query parameters
 			const { egoToken } = context;
 			const updates = pickBy(get(args, 'updates', {}), (v) => v !== undefined);
-			const shortName = get(args, 'shortName', {});
-
+			const programShortName = get(args, 'shortName', {});
+			const dataCenterShortName = args.updates.dataCenterShortName;
+			delete updates.dataCenterShortName;
```

Comment:
could assign the var `updates` above exactly how you want it once, instead of deleting a property.
using `delete` can lead to functionality being hard to follow vs defined once and not mutated

Reply:
Yes these are Daniel's original changes, I will revisit this once the original branch is merged.
One issue is `updates` is not Typed, so I figured `delete` was a simple way to accomplish what's being done here, vs making a method for duplicating an object minus one key

Reply:
ah gotcha. Yeah there's not a whole lot of typing to assist.
I wouldn't worry about making a method.. there's plenty of lodash util use already, you could just do an `omit`. returns a new object.

i'm not actually sure why pickby is in use either to be honest. Looks like it's just checking for undefined values but I thought that's enforced by the gql endpoints because of it's typing ?

Reply:
See how this looks instead? https://github.com/icgc-argo/platform-api/pull/714/commits/8d3f99dae9e966c650a5dc8a2d701714d4d711c6

I checked, there is no use of the Data Center on Platform UI yet, so we should be able to change the types around for now